lib/jbuilder/schema/template.rb

```diff
@@ -254,5 +261,15 @@ def _merge_block(key)
       value = _object(value, _required!(value.keys)) unless value[:type] == :array || value.key?(:$ref)
       _merge_values(current_value, value)
     end
+
+    def _within_block?(&block)
+      block.present? && _one_line?(block.source)
```

Comment:
iirc `source` comes from the `method_source` gem that Rails bundled, but maybe 7.1 doesn't bundle it anymore?

Though I'm wondering if we could the stacktrace via `caller_locations` to detect if we're within a block instead?